From 836516d8150d0b0642dade2ca5d8d3dcbd6f2eab Mon Sep 17 00:00:00 2001
From: "Gryska, ArkadiuszX" <arkadiuszx.gryska@intel.com>
Date: Thu, 27 Mar 2025 15:59:23 +0100
Subject: [PATCH] [PATCH 2/4] ice: set ICE_SCHED_DFLT_BURST_SIZE to 2048

Signed-off-by: Gryska, ArkadiuszX <arkadiuszx.gryska@intel.com>
---
 src/ice_type.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ice_type.h b/src/ice_type.h
index c1a749e..22bca3f 100644
--- a/src/ice_type.h
+++ b/src/ice_type.h
@@ -1091,7 +1091,7 @@ enum ice_rl_type {
 #define ICE_SCHED_DFLT_RL_PROF_ID	0
 #define ICE_SCHED_NO_SHARED_RL_PROF_ID	0xFFFF
 #define ICE_SCHED_INVAL_PROF_ID		0xFFFF
-#define ICE_SCHED_DFLT_BURST_SIZE	(15 * 1024)	/* in bytes (15k) */
+#define ICE_SCHED_DFLT_BURST_SIZE	(2 * 1024)	/* in bytes (2k) */
 
 /* Access Macros for Tx Sched RL Profile data */
 #define ICE_TXSCHED_GET_RL_PROF_ID(p) le16_to_cpu((p)->info.profile_id)
-- 
2.42.0.windows.1

