From 4eba953cebe01eaabe16ce50119c7d969938065d Mon Sep 17 00:00:00 2001
From: Frank Du <frank.du@intel.com>
Date: Fri, 26 Jan 2024 16:23:34 +0800
Subject: [PATCH] avdevice: add mtl in/out dev support

Signed-off-by: Frank Du <frank.du@intel.com>
---
 configure                | 9 +++++++++
 libavdevice/Makefile     | 6 ++++++
 libavdevice/alldevices.c | 9 +++++++++
 3 files changed, 24 insertions(+)

diff --git a/configure b/configure
index ca184072b3..b7582d0f39 100755
--- a/configure
+++ b/configure
@@ -309,6 +309,7 @@ External library support:
                            if openssl, gnutls or libtls is not used [no]
   --enable-mediacodec      enable Android MediaCodec support [no]
   --enable-mediafoundation enable encoding via MediaFoundation [auto]
+  --enable-mtl             enable Media transport libraray support for st2110 [no]
   --disable-metal          disable Apple Metal framework [autodetect]
   --enable-libmysofa       enable libmysofa, needed for sofalizer filter [no]
   --enable-openal          enable OpenAL 1.1 capture support [no]
@@ -1908,6 +1909,7 @@ EXTERNAL_LIBRARY_LIST="
     libzvbi
     lv2
     mediacodec
+    mtl
     openal
     opengl
     openssl
@@ -3597,6 +3599,12 @@ kmsgrab_indev_deps="libdrm"
 lavfi_indev_deps="avfilter"
 libcdio_indev_deps="libcdio"
 libdc1394_indev_deps="libdc1394"
+mtl_st20p_indev_deps="mtl"
+mtl_st20p_outdev_deps="mtl"
+mtl_st22p_indev_deps="mtl"
+mtl_st22p_outdev_deps="mtl"
+mtl_st30p_indev_deps="mtl"
+mtl_st30p_outdev_deps="mtl"
 openal_indev_deps="openal"
 opengl_outdev_deps="opengl"
 opengl_outdev_suggest="sdl2"
@@ -6873,6 +6881,7 @@ enabled mmal              && { check_lib mmal interface/mmal/mmal.h mmal_port_co
                                  check_lib mmal interface/mmal/mmal.h mmal_port_connect -lmmal_core -lmmal_util -lmmal_vc_client -lbcm_host; } ||
                                die "ERROR: mmal not found" &&
                                check_func_headers interface/mmal/mmal.h "MMAL_PARAMETER_VIDEO_MAX_NUM_CALLBACKS"; }
+enabled mtl               && require_pkg_config mtl "mtl >= 22.12.0" mtl/st_pipeline_api.h mtl_init
 enabled openal            && { check_pkg_config openal "openal >= 1.1" "AL/al.h" alGetError ||
                                { for al_extralibs in "${OPENAL_LIBS}" "-lopenal" "-lOpenAL32"; do
                                check_lib openal 'AL/al.h' alGetError "${al_extralibs}" && break; done } ||
diff --git a/libavdevice/Makefile b/libavdevice/Makefile
index c30449201d..23b0e130fe 100644
--- a/libavdevice/Makefile
+++ b/libavdevice/Makefile
@@ -34,6 +34,12 @@ OBJS-$(CONFIG_IEC61883_INDEV)            += iec61883.o
 OBJS-$(CONFIG_JACK_INDEV)                += jack.o timefilter.o
 OBJS-$(CONFIG_KMSGRAB_INDEV)             += kmsgrab.o
 OBJS-$(CONFIG_LAVFI_INDEV)               += lavfi.o
+OBJS-$(CONFIG_MTL_ST20P_INDEV)           += mtl_common.o mtl_st20p_rx.o
+OBJS-$(CONFIG_MTL_ST20P_OUTDEV)          += mtl_common.o mtl_st20p_tx.o
+OBJS-$(CONFIG_MTL_ST22P_INDEV)           += mtl_common.o mtl_st22p_rx.o
+OBJS-$(CONFIG_MTL_ST22P_OUTDEV)          += mtl_common.o mtl_st22p_tx.o
+OBJS-$(CONFIG_MTL_ST30P_INDEV)           += mtl_common.o mtl_st30p_rx.o
+OBJS-$(CONFIG_MTL_ST30P_OUTDEV)          += mtl_common.o mtl_st30p_tx.o
 OBJS-$(CONFIG_OPENAL_INDEV)              += openal-dec.o
 OBJS-$(CONFIG_OPENGL_OUTDEV)             += opengl_enc.o
 OBJS-$(CONFIG_OSS_INDEV)                 += oss_dec.o oss.o
diff --git a/libavdevice/alldevices.c b/libavdevice/alldevices.c
index 8a90fcb5d7..b183c18e9c 100644
--- a/libavdevice/alldevices.c
+++ b/libavdevice/alldevices.c
@@ -40,6 +40,15 @@ extern const AVInputFormat  ff_iec61883_demuxer;
 extern const AVInputFormat  ff_jack_demuxer;
 extern const AVInputFormat  ff_kmsgrab_demuxer;
 extern const AVInputFormat  ff_lavfi_demuxer;
+extern const AVInputFormat  ff_mtl_st20p_demuxer;
+extern const FFOutputFormat ff_mtl_st20p_muxer;
+extern const AVInputFormat  ff_mtl_st22_demuxer;
+extern const AVInputFormat  ff_mtl_st22p_demuxer;
+extern const FFOutputFormat ff_mtl_st22_muxer;
+extern const FFOutputFormat ff_mtl_st22p_muxer;
+extern const AVInputFormat  ff_mtl_st30p_demuxer;
+extern const FFOutputFormat ff_mtl_st30p_muxer;
+extern const FFOutputFormat ff_mtl_st30p_pcm16_muxer;
 extern const AVInputFormat  ff_openal_demuxer;
 extern const FFOutputFormat ff_opengl_muxer;
 extern const AVInputFormat  ff_oss_demuxer;
-- 
2.34.1

