From c98cf508b2a7556b916f555f493398f70fc8c916 Mon Sep 17 00:00:00 2001
From: Ric Li <ming3.li@intel.com>
Date: Wed, 10 May 2023 17:37:11 +0800
Subject: [PATCH] Temporarily disable pcapng build

---
 lib/pcapng/meson.build | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/lib/pcapng/meson.build b/lib/pcapng/meson.build
index 4549925d41..da938bbcb7 100644
--- a/lib/pcapng/meson.build
+++ b/lib/pcapng/meson.build
@@ -1,6 +1,12 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright(c) 2019 Microsoft Corporation
 
+if is_windows
+    build = false
+    reason = 'not supported on Windows'
+    subdir_done()
+endif
+
 sources = files('rte_pcapng.c')
 headers = files('rte_pcapng.h')
 
-- 
2.39.1.windows.1

