From 12c829262855f516656c41d7232302656ecb711e Mon Sep 17 00:00:00 2001
From: Szumski Tomasz <tomasz.szumski@intel.com>
Date: Fri, 11 Jul 2025 13:37:08 +0200
Subject: [PATCH 4/4] version: update to Kahawai_2.2.8

---
 src/ice_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ice_main.c b/src/ice_main.c
index 4033e5e..7213063 100644
--- a/src/ice_main.c
+++ b/src/ice_main.c
@@ -35,7 +35,7 @@
 #define DRV_VERSION_MINOR 2
 #define DRV_VERSION_BUILD 8
 
-#define DRV_VERSION	"2.2.8"
+#define DRV_VERSION	"Kahawai_2.2.8"
 #define DRV_SUMMARY	"Intel(R) Ethernet Connection E800 Series Linux Driver"
 #ifdef ICE_ADD_PROBES
 #define DRV_VERSION_EXTRA "_probes"
-- 
2.34.1

